pages/_app.tsx

```diff
@@ -1,29 +1,51 @@
 import 'tailwindcss/tailwind.css';
-import { createRef, useState } from 'react';
+import { createRef, useCallback, useEffect, useRef, useState } from 'react';
 import { AppProps } from 'next/app';
+import Hls from 'hls.js';
 
 import PlayerControlPanel from '@/components/playerControlPanel';
 import { ChatWrapper } from '@/components/ShoutBox/shoutbox';
 import VideoPlayer from '@/components/videoPlayer';
 import { ShoutBoxAndVideoProvider } from '@/hooks/useShoutboxAndVideo';
 
+const isHlsLive = process.env.NEXT_PUBLIC_HLS_MODE === 'live';
+
 const AUDIO_STREAM_URL = 'https://player.turunwappuradio.com/wappuradio.mp3';
+const HLS_STREAM_URL = 'https://stream.turunwappuradio.com/twr_chunklist.m3u8';
 
 const MyApp = ({ Component, pageProps }: AppProps) => {
   const audioEl = createRef<HTMLAudioElement>();
+  const hls = useRef<Hls | null>(null);
   const [playing, setPlaying] = useState(false);
   const [muted, setMuted] = useState(false);
   const [volume, setVolume] = useState(0.8);
   const [playClicked, setPlayClicked] = useState(false);
 
+  const loadAudioStream = useCallback(() => {
+    if (Hls.isSupported() && isHlsLive) {
+      hls.current = new Hls();
+      hls.current.loadSource(HLS_STREAM_URL);
+      hls.current.attachMedia(audioEl.current);
+    } else if (
+      audioEl.current &&
+      audioEl.current.canPlayType('application/vnd.apple.mpegurl')
+    ) {
+      audioEl.current.src = AUDIO_STREAM_URL;
+    }
+  }, [audioEl]);
+
+  // Load the stream on initial page load.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(loadAudioStream, []);
+
   const handlePlayPause = () => {
     setPlayClicked(true);
 
     if (audioEl.current.paused) audioEl.current.play();
     else {
       // Pause, but then load the stream again ready to start
       audioEl.current.pause();
-      audioEl.current.src = AUDIO_STREAM_URL;
+      loadAudioStream();
```

Comment:
Hm lataakos tää mahdollisesti väärän chunkin muistiin? Jos pauseen, odottelen tunnin, pistän päälle nii osaako jotenkin hakee sen uusimman vai halutaanko nimenomaan, et jatkaa vanhasta kohdasta? 

Vanhassa playerissä tuli itelle vastaan joskus tää sama, et vanha kohta jäi muistiin (which is nice), mutta hetken päästä se kuitenkin hyppäs nykyhetkeen. Miten mahtaa toimia ja halutaan toimivan?

Reply:
Hmm, hyvä huomio. Oiskohan tää parempi jos `loadAudioStream` kutsuttaisiin play:n painamisessa, eikä pausen? Voisin testata tän.